experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1420,31 +1423,70 @@ def _validate_required_or_excluded_experiments(self, value):
     ):
-        if schema := feature_config.schemas.get(version=None).schema:
-            json_schema = json.loads(schema)
-            json_value = json.loads(value)
+        schemas_in_range = feature_config.get_versioned_schema_range(
+            min_version, max_version
+        )
+        if schemas_in_range.unsupported_in_range:
+            return [
+                NimbusConstants.ERROR_FEATURE_CONFIG_UNSUPPORTED_IN_RANGE.format(
+                    feature_config=feature_config,
+                )
+            ]
 
-            if not localizations:
-                return self._validate_schema(json_value, json_schema)
+        errors = []
 
-            for locale_code, substitutions in localizations.items():
-                try:
-                    substituted_value = self._substitute_localizations(
-                        json_value, substitutions, locale_code
-                    )
-                except LocalizationError as e:
-                    return [str(e)]
+        for version in schemas_in_range.unsupported_versions:
+            errors.append(
+                NimbusConstants.ERROR_FEATURE_CONFIG_UNSUPPORTED_IN_VERSION.format(
+                    feature_config=feature_config,
+                    version=version,
+                )
+            )
+
+        json_value = json.loads(value)
+
+        for schema in schemas_in_range.schemas:
+            if schema.schema is None:  # Only in tests.
+                continue
 
-                if schema_errors := self._validate_schema(substituted_value, json_schema):
-                    return [
-                        (
+            json_schema = json.loads(schema.schema)
+            if not localizations:
+                errors.extend(
+                    self._validate_schema(json_value, json_schema, schema.version)
+                )
+            else:
+                for locale_code, substitutions in localizations.items():
+                    try:
+                        substituted_value = self._substitute_localizations(
+                            json_value, substitutions, locale_code
+                        )
+                    except LocalizationError as e:
+                        errors.append(str(e))
+                        continue
+
+                    if schema_errors := self._validate_schema(
+                        substituted_value, json_schema, schema.version
+                    ):
+                        err_msg = (
                             f"Schema validation errors occured during locale "
                             f"substitution for locale {locale_code}"
-                        ),
-                        *schema_errors,
-                    ]
+                        )
+
+                        if schema.version is not None:
+                            err_msg += f" at version {schema.version}"
+
+                        errors.append(err_msg)
+                        errors.extend(schema_errors)
+
+        if errors:
```

Comment:
Oh I see, we start with the empty list and expect to be able to append/extend, does it have to be `None` down here instead of `[]`?

Reply:
It doesn't *have* to be

Reply:
Yeah it's just something to think about, if something could be X | None you always have to check `is not None` when you get the response, whereas if it's a type that has a falsy value then you can still use that as a signal that there's no data without needing to worry about whether it's X | None.